src/Sentry/SentryLog.cs

```diff
@@ -0,0 +1,281 @@
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Internal;
+using Sentry.Protocol;
+
+namespace Sentry;
+
+/// <summary>
+/// Represents the Sentry Log protocol.
+/// <para>This API is experimental and it may change in the future.</para>
+/// </summary>
+[Experimental(DiagnosticId.ExperimentalFeature)]
+public sealed class SentryLog : ISentryJsonSerializable
+{
+    private readonly Dictionary<string, SentryAttribute> _attributes;
+
+    [SetsRequiredMembers]
+    internal SentryLog(DateTimeOffset timestamp, SentryId traceId, SentryLogLevel level, string message)
+    {
+        Timestamp = timestamp;
+        TraceId = traceId;
+        Level = level;
+        Message = message;
+        // 7 is the number of built-in attributes, so we start with that.
+        _attributes = new Dictionary<string, SentryAttribute>(7);
+    }
+
+    /// <summary>
+    /// The timestamp of the log.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    /// <remarks>
+    /// Sent as seconds since the Unix epoch.
+    /// </remarks>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
```

Comment:
Possibly overkill decorating every property with this... the class is already decorated with `Experimental`.

I think it's enough if we put the option to enable this feature under `SentryOptions.Experimenta` and then decorate any public methods in the API (the stuff on `IHub` I'm guessing).

Potentially we could decorate public classes like this as well, although I can't see anybody using these classes independently of the experimental APIs they support.

Reply:
`SentryLog` is used in callbacks, and when user code utilizes implicit typing, no warning would appear when invoking any member.

But yes ... I still need to move the Options to the experimental section, as the Attribute does not work for our .NET Standard nor .NET Framework targets.